### PR TITLE
DDF 3702 - Pre-populated Query Templates in the UI

### PR DESCRIPTION
**Pull request by middlets719**
#### What does this PR do?
Implements the default values for templates on the UI
#### Who is reviewing it? 
@Schachte @adimka 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@bdeining
@jlcsmith
#### How should this be tested? (List steps with links to updated documentation)
Reinstall DDF instance.  
Prior to initial startup add the etc/forms directory to the DDF instance
Edit the XML files to include a default value
  Default values are loaded into the first Literal Tag after the function Tag.  Ex:
            <fes:Function name="template.value.v1">
                <fes:Literal>Connexta</fes:Literal> (DefaultValue)
Start DDF
Open the template that was modified to include a default
Verify default value is displayed in the valid fields
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-3702
#### Screenshots (if appropriate)
image

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js [hunk 1]
```diff
@@ -126,6 +126,9 @@ define([
             }
             return value;
         },
+        transformValue: function (value, defaultValue, comparator) {
```
Comment (Schachte) [inline ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js]: Do we need 3 params in this function if you only ever use `value`?

File: ui/packages/catalog-ui-search/src/main/webapp/component/filter/search-form/filter.search-form.view.js [hunk 2]
```diff
@@ -59,12 +66,14 @@ define([
                 case 'DWITHIN':
                     break;
                 default:
-                   if (value[0] == null) {
-                        value[0] = "";
-                   } else if (value[0].constructor === Object) {
-                        value[0] = value[0].value;
-                   }
-                   break;
+					if (value[0] != null && value[0].constructor === Object) {
```
Comment (Schachte) [inline ui/packages/catalog-ui-search/src/main/webapp/component/filter/search-form/filter.search-form.view.js]: Indentation needs to be fixed

**Comment by Schachte** [comment]
Might be good to squash commits

File: ui/packages/catalog-ui-search/src/main/webapp/component/filter/search-form/filter.search-form.view.js [hunk 3]
```diff
@@ -46,6 +46,13 @@ define([
             this.filterAttribute.currentView.turnOffEditing();
         },
         transformValue: function (value, comparator) {
+            var defaultValue;
+            if(typeof this.model.get('defaultValue') !== 'undefined'){
+                defaultValue = Common.duplicate(this.model.get('defaultValue'));
+            }
+            else{
```
Comment (Schachte) [inline ui/packages/catalog-ui-search/src/main/webapp/component/filter/search-form/filter.search-form.view.js]: Remove this else block

File: ui/packages/catalog-ui-search/src/main/webapp/component/filter/search-form/filter.search-form.view.js [hunk 4]
```diff
@@ -46,6 +46,13 @@ define([
             this.filterAttribute.currentView.turnOffEditing();
         },
         transformValue: function (value, comparator) {
+            var defaultValue;
```
Comment (Schachte) [inline ui/packages/catalog-ui-search/src/main/webapp/component/filter/search-form/filter.search-form.view.js]: Change to defaultValue = "";

**Comment by Schachte** [comment]
LG2M after above comments addressed

File: ui/packages/catalog-ui-search/src/main/webapp/component/filter/search-form/filter.search-form.view.js [hunk 5]
```diff
@@ -77,6 +86,27 @@ define([
                 : this.filterInput.currentView.model;
             property.set('isEditing', true);
         },
+        setFilter: function(filter){
+            setTimeout(function(){
```
[the remark this answers is not in the document]
Comment (bdeining) [reply ui/packages/catalog-ui-search/src/main/webapp/component/filter/search-form/filter.search-form.view.js]: This is copied from the parent class; based on my understanding of it, the setTimeout here puts this method at the top of the queue to be called

File: ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js (hunk 1 above)
Comment (middlets719) [reply ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js]: Removed, this was here on accident

**Comment by middlets719** [comment]
build now

**Comment by Schachte** [comment]
🎉 Was able to add default textual values to first literal in filter XML and see it pre-populate the text fields on the UI.

**Comment by bdeining** [comment]
build now

**Comment by jlcsmith** [comment]
build now

**Pull request merged**